chainerx_cc/chainerx/cuda/cuda_device/linalg.cu

```diff
@@ -204,5 +283,143 @@ public:
 
 CHAINERX_CUDA_REGISTER_KERNEL(InverseKernel, CudaInverseKernel);
 
+class CudaSVDKernel : public SVDKernel {
+public:
+    std::tuple<Array, Array, Array> Call(const Array& a, bool full_matrices, bool compute_uv) override {
+        Device& device = a.device();
+        Dtype dtype = a.dtype();
+        CudaSetDeviceScope scope{device.index()};
+
+        CHAINERX_ASSERT(a.ndim() == 2);
+
+        int64_t n = a.shape()[0];
+        int64_t m = a.shape()[1];
+
+        Array x{};
```

Comment:
Elements (n, m) can be allocated here instead of being delayed as the properties are already known.